File: c_language_solutions/118. Pascal's Triangle.c
int** generate(int numRows, int* returnSize, int** returnColumnSizes) {
    if (numRows <= 0) {
        *returnSize = 0;
        *returnColumnSizes = NULL;
        return NULL;
    }

    int** result = (int**)malloc(numRows * sizeof(int*));
    *returnColumnSizes = (int*)malloc(numRows * sizeof(int));
    *returnSize = numRows;

    for (int i = 0; i < numRows; i++) {
        result[i] = (int*)malloc((i + 1) * sizeof(int));
        (*returnColumnSizes)[i] = i + 1;

        result[i][0] = 1;
        result[i][i] = 1;

        for (int j = 1; j < i; j++) {
            result[i][j] = result[i - 1][j - 1] + result[i - 1][j];
        }
    }

    return result;
}

File: python_codes/118. Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        nums = []  
        for i in range(numRows):
            nums.append([0] * (i + 1))
            nums[i][0], nums[i][i] = 1, 1 
            for j in range(1, i):
                nums[i][j] = nums[i-1][j-1] + nums[i-1][j]
        return nums
